### DISCUSSION TITLE
How to revert back to Android ?

### DISCUSSION THREAD

**Discussion by Kullsputra** [devmfc/debian-on-amlogic#90] [General]
Anybody know how to revert back to Android after installing debian-on-amlogic to emmc (and enabling multi-boot with the script) ?

Seems like the bootloader was changed in ways that prevent my computer from seeing the tvbox and so I'm unable to use Aml USB Burn Tool to restore the original firmware.  Maybe I can flash the stock firmware .img file to the emmc from debian with something like dd ? 

Already tried shorting the emmc appropriate pins, the reset A/V pin, nothing seems to work.

Any help would be appreciated, thanks.

**Comment by devmfc** [comment]
- What box do you have?
- Are you sure you had stock u-boot on your box? I have never seen that the multiboot script interfered with just using the reset button to enable USB Burn Tool.  I think you had some other image flashed before (slimbox or something?)?
- Do you use the right USB port? Can you try the other ports as well?
- To reverse multiboot you only have to replace the u-boot environment value *bootcmd* with the value in *cmd_boot_org*. Can you post the output of `fw_printenv`?

My images DO NOT replace the bootloader, which is responsible for the USB Burn connection, so the bootloader should be fine.

You can not flash just any image to emmc, so I won't go there if you don't know what you are doing.

Trying to short the emmc pins should only be needed as a last resort.

**Comment by Kullsputra** [comment]
The box is a X96 X4, sorry for not mentioning it earlier.
I have ever only used the stock rom, booted CoreELEC from a usb flash drive once. Just heard about slimbox for S96 X4 recently.
Tried both usb ports, 2.0 and 3.0.
A backup was made of the original u-boot (one of the scripts, I'm guessing), I can see the file in /root, I think. Any way I can use that?
I'll try and post the output of fw_printenv in here later, can't do it right now.
About the bootloader, it was just a guess based on what I was reading while researching the forums about the subject.
I don't really know what happened. I can plug a usb drive with CoreELEC and it'll boot normally and without using the reset pin, if I leave the usb ports empty then debian will boot, but no device is show when I plug the box in the computer using a male-to-male USB cable, even using the A/V pin. Also tried 2 different cables.
Maybe I can prepare a usb drive to flash the stock android firmware to it? If it runs CoreELEC maybe it can run some script in the usb drive. 
At the moment I'm not using the box as I don't have much use for a tvbox server right now, so I either revert back to Android so it can be useful as a portable media/streaming device or given to someone, or it'll sit in some drawer. Wasteful but I don't really care if it's trashed or bricked permanently at this point. I'll try and connect it to another pc and see if it shows up.

Thanks for answering.

**Reply by devmfc** [reply]
You bootloader is not likely to be the problem. My images don't replace it, CoreELEC does not replace it either. So either your bootloader never worked with the USB Burn protocol (or more likely, does not read the state of the reset button). Or the u-boot environment is not updated correctly by my script or the CoreELEC script (or the combination of the two).

For X96X4 you should use USB3 port AND you should use the newer USB Burn Tool (V3.20 or higher), because the older versions for older SoC's will not work. When you connect the USB A-A cable, you should also press and hold the reset button until it shows in USB Burn Tool

If you try to replace the bootloader without really having the right bootloader there is a great risk you will (soft) brick your box.

You really need the output of fw_printenv, or even better, serial output while booting and pressing the reset button.

**Comment by Kullsputra** [comment]
I'll try your suggestions. Funny thing is I have used USB Burn Tool before in this box and it worked no problem, it was even a old version as well. Can I read the serial ouput with a arduino board? I may have a ch340 in here somewhere, need to take a look.

Here you go:

1080p60hz_deepcolor=444,8bit
1080p_h=1080
1080p_w=1920
1080p_x=0
1080p_y=0
2160p60hz_deepcolor=420,10bit
2160p_h=2160
2160p_w=3840
2160p_x=0
2160p_y=0
576cvbs_deepcolor=444,8bit
576cvbs_h=576
576cvbs_w=720
576cvbs_x=0
576cvbs_y=0
EnableSelinux=permissive
Irq_check_en=0
active_slot=_a
aml_dt=sc2_s905x4_ah212-4g
arch=arm
avb2=1
baudrate=115200
bcb_cmd=get_avb_mode;get_valid_slot;if test ${vendor_boot_mode} = true; then setenv loadaddr_kernel 0x3080000;setenv dtb_mem_addr 0x1000000;fi;if test ${active_slot} != normal; then echo ab mode, read dtb from kernel;setenv common_dtb_load imgread dtb ${boot_part} ${dtb_mem_addr};fi;
bestdolbyvision=true
board=ohm
board_name=sc2_ah212_ddr3
boot_ddr_type=DDR3
boot_emmc_parts=1 2 1c 1b
boot_part=boot_a
boot_script_order=bootscript cfgload s905_autoscript boot.scr 
bootargs=init=/init console=ttyS0,921600 no_console_suspend earlycon=aml-uart,0xfe07a000 ramoops.pstore_en=1 ramoops.record_size=0x8000 ramoops.console_size=0x4000 loop.max_part=4 loglevel={loglevel} otg_device=0 logo=osd0,loaded,0x00300000 vout=1080p60hz,disable panel_type=lcd_1 hdmitx=,444,8bit
 hdmimode=1080p60hz hdmichecksum=0x3ff20000 dolby_vision_on=0 hdr_policy=1 hdr_priority= frac_rate_policy=1 hdmi_read_edid=1 cvbsmode=576cvbs osd_reverse=0 video_reverse=0 irq_check_en=0 androidboot.selinux=permissive androidboot.firstboot=0 jtag=disable androidboot.hardware=amlogic mac=90:0E:B3:22:2F:48 androidboot.mac=90:0E:B3:22:2F:48 androidboot.wificountrycode=US androidboot.serialno=1234567890
bootcmd=run cmd_boot_get_order; for var in "${boot_order}"; do run cmd_boot_${var}; done; run cmd_boot_org
bootdelay=0
bootfromemmc=run cfgloademmc
bootfromnand=0
bootfromsd=if mmcinfo; then run cfgloadsd; if fatload mmc 0 ${loadaddr} kernel.img; then run sddtb; setenv bootargs ${bootargs} bootfromsd; bootm; fi; fi
bootfromusb=usb start 0; run cfgloadusb; if fatload usb 0 ${loadaddr} kernel.img; then run usbdtb; setenv bootargs ${bootargs} bootfromusb; bootm; fi
bootm_low=0
bootm_size=f1000000
bootup_offset=0x13a6d8
bootup_size=0x3f4848
ce_on_emmc=no
cfgloademmc=for p in 1 2 3 4 5 6 7 8 9 A B C D E F 10 11 12 13 14 15 16 17 18; do if fatload mmc 1:${p} ${loadaddr} cfgload; then setenv device mmc; setenv devnr 1; setenv partnr ${p}; setenv ce_on_emmc "yes"; autoscr ${loadaddr}; fi; done;
cfgloadsd=if fatload mmc 0:1 ${loadaddr} cfgload; then setenv device mmc; setenv devnr 0; setenv partnr 1; autoscr ${loadaddr}; fi
cfgloadusb=if fatload usb 0:1 ${loadaddr} cfgload; then setenv device usb; setenv devnr 0; setenv partnr 1; autoscr ${loadaddr}; fi
cmd_boot_emmc=setenv devtype mmc; setenv devnum 1; for partnum in "${boot_emmc_parts}"; do setenv distro_bootpart ${partnum}; run cmd_bootsript_load; done
cmd_boot_get_order=boot_order="${default_boot_order}"; if test "${onetime_boot_order}" != "" ; then boot_order=${onetime_boot_order}; setenv onetime_boot_order; saveenv; fi; if test "${bootfromnand}" = "1"; then setenv bootfromnand 0; saveenv; boot_order="org ${boot_order}"; fi; echo "resulting boot_order:${boot_order}";
cmd_boot_org=1080p60hz_deepcolor=444,8bit
1080p_h=1080
1080p_w=1920
1080p_x=0
1080p_y=0
2160p60hz_deepcolor=420,10bit
2160p_h=2160
2160p_w=3840
2160p_x=0
2160p_y=0
576cvbs_deepcolor=444,8bit
576cvbs_h=576
576cvbs_w=720
576cvbs_x=0
576cvbs_y=0
EnableSelinux=permissive
Irq_check_en=0
active_slot=_a
aml_dt=sc2_s905x4_ah212-4g
arch=arm
avb2=1
baudrate=115200
bcb_cmd=get_avb_mode;get_valid_slot;if test ${vendor_boot_mode} = true; then setenv loadaddr_kernel 0x3080000;setenv dtb_mem_addr 0x1000000;fi;if test ${active_slot} != normal; then echo ab mode, read dtb from kernel;setenv common_dtb_load imgread dtb ${boot_part} ${dtb_mem_addr};fi;
bestdolbyvision=true
board=ohm
board_name=sc2_ah212_ddr3
boot_ddr_type=DDR3
boot_emmc_parts=1 2 1c 1b
boot_part=boot_a
boot_script_order=bootscript cfgload s905_autoscript boot.scr 
bootargs=init=/init console=ttyS0,921600 no_console_suspend earlycon=aml-uart,0xfe07a000 ramoops.pstore_en=1 ramoops.record_size=0x8000 ramoops.console_size=0x4000 loop.max_part=4 loglevel={loglevel} otg_device=0 logo=osd0,loaded,0x00300000 vout=1080p60hz,disable panel_type=lcd_1 hdmitx=,444,8bit
 hdmimode=1080p60hz hdmichecksum=0x23560000 dolby_vision_on=0 hdr_policy=1 hdr_priority= frac_rate_policy=1 hdmi_read_edid=1 cvbsmode=576cvbs osd_reverse=0 video_reverse=0 irq_check_en=0 androidboot.selinux=permissive androidboot.firstboot=0 jtag=disable androidboot.hardware=amlogic mac=90:0E:B3:22:2F:48 androidboot.mac=90:0E:B3:22:2F:48 androidboot.wificountrycode=US androidboot.serialno=1234567890
bootcmd=if test ${bootfromnand} = 1; then setenv bootfromnand 0; saveenv; else run bootfromsd; run bootfromusb; run bootfromemmc; fi; run storeboot
bootdelay=0
bootfromemmc=run cfgloademmc
bootfromnand=0
bootfromsd=if mmcinfo; then run cfgloadsd; if fatload mmc 0 ${loadaddr} kernel.img; then run sddtb; setenv bootargs ${bootargs} bootfromsd; bootm; fi; fi
bootfromusb=usb start 0; run cfgloadusb; if fatload usb 0 ${loadaddr} kernel.img; then run usbdtb; setenv bootargs ${bootargs} bootfromusb; bootm; fi
bootm_low=0
bootm_size=f1000000
bootup_offset=0x13a6d8
bootup_size=0x3f4848
ce_on_emmc=no
cfgloademmc=for p in 1 2 3 4 5 6 7 8 9 A B C D E F 10 11 12 13 14 15 16 17 18; do if fatload mmc 1:${p} ${loadaddr} cfgload; then setenv device mmc; setenv devnr 1; setenv partnr ${p}; setenv ce_on_emmc "yes"; autoscr ${loadaddr}; fi; done;
cfgloadsd=if fatload mmc 0:1 ${loadaddr} cfgload; then setenv device mmc; setenv devnr 0; setenv partnr 1; autoscr ${loadaddr}; fi
cfgloadusb=if fatload usb 0:1 ${loadaddr} cfgload; then setenv device usb; setenv devnr 0; setenv partnr 1; autoscr ${loadaddr}; fi
cmd_boot_emmc=setenv devtype mmc; setenv devnum 1; for partnum in "${boot_emmc_parts}"; do setenv distro_bootpart ${partnum}; run cmd_bootsript_load; done
cmd_boot_get_order=boot_order="${default_boot_order}"; if test "${onetime_boot_order}" != "" ; then boot_order=${onetime_boot_order}; setenv onetime_boot_order; saveenv; fi; if test "${bootfromnand}" = "1"; then setenv bootfromnand 0; saveenv; boot_order="org ${boot_order}"; fi; echo "resulting boot_order:${boot_order}";
cmd_boot_sd=if mmcinfo; then setenv devtype mmc; setenv devnum 0; setenv distro_bootpart 1; run cmd_bootsript_load; fi
cmd_boot_usb=if usb storage; then ; else usb start; fi; setenv devtype usb; setenv distro_bootpart 1; for devnum in "0 1 2 3"; do setenv devnum ${devnum}; run cmd_bootsript_load; done
cmd_bootsript_load=for scrpt in "${boot_script_order}"; do fatload ${devtype} ${devnum}:${distro_bootpart} ${loadaddr} ${scrpt} && run cmd_make_compat && autoscr ${loadaddr}; done;
cmd_make_compat=cmd_read="fatload ${devtype} ${devnum}:${distro_bootpart}"; device=${devtype};devnr=${devnum};partnr=${distro_bootpart};partnum=${distro_bootpart};autoscript_source=${devtype}
cmdline_keys=setenv usid 1234567890; setenv region_code US;if keyman init 0x1234; then if keyman read usid ${loadaddr} str; then fi;if keyman read region_code ${loadaddr} str; then fi;if keyman read mac ${loadaddr} str; then setenv bootargs ${bootargs} mac=${mac} androidboot.mac=${mac};fi;if keyman read deviceid ${loadaddr} str; then setenv bootargs ${bootargs} androidboot.deviceid=${deviceid};fi;fi;setenv bootargs ${bootargs} androidboot.wificountrycode=${region_code};setenv bootargs ${bootargs} androidboot.serialno=${usid};setenv serial ${usid}; setenv serial# ${usid};factory_provision init;
colorattribute=444,8bit
common_dtb_load=imgread dtb boot_a 0x1000000
cpu=armv8
cvbs_drv=0
cvbsmode=576cvbs
default_boot_order=sd usb emmc org
display_bpp=16
display_color_bg=0
display_color_fg=0xffff
display_color_index=16
display_height=1080
display_layer=osd0
display_width=1920
dolby_status=0
dolby_vision_on=0
dtb_mem_addr=0x1000000
eth6addr=90:0e:b3:22:2f:48
fatload_dev=usb
fb_addr=0x00300000
fb_height=1080
fb_width=1920
fdt_high=0x20000000
fdtaddr=1000000
fdtcontroladdr=f077c700
firstboot=0
frac_rate_policy=1
fs_type=rootfstype=ramfs
gatewayip=10.18.9.1
get_os_type=if store read ${os_ident_addr} ${boot_part} 0 0x1000; then os_ident ${os_ident_addr}; fi
hdmi_colordepth=8
hdmi_colorspace=auto
hdmi_read_edid=1
hdmichecksum=0x23560000
hdmimode=1080p60hz
hdr_policy=1
hostname=arm_gxbb
init_display=get_rebootmode;echo reboot_mode:::: ${reboot_mode};if test ${reboot_mode} = quiescent; then setenv reboot_mode_android quiescent;setenv dolby_status 0;setenv dolby_vision_on 0;run storeargs;setenv bootargs ${bootargs} androidboot.quiescent=1;osd open;osd clear;else if test ${reboot_mode} = recovery_quiescent; then setenv reboot_mode_android quiescent;setenv dolby_status 0;setenv dolby_vision_on 0;run storeargs;setenv bootargs ${bootargs} androidboot.quiescent=1;osd open;osd clear;else setenv reboot_mode_android normal;run storeargs;hdmitx hpd;hdmitx get_preferred_mode;hdmitx get_parse_edid;setenv dolby_status 0;setenv dolby_vision_on 0;osd open;osd clear;imgread pic logo bootup $loadaddr;bmp display $bootup_offset;bmp scale;vout output ${outputmode};vpp hdrpkt;fi;fi;
initargs=init=/init console=ttyS0,921600 no_console_suspend earlycon=aml-uart,0xfe07a000 ramoops.pstore_en=1 ramoops.record_size=0x8000 ramoops.console_size=0x4000 loop.max_part=4 
initrd_high=df800000
ipaddr=10.18.9.97
is.bestmode=false
jtag=disable
loadaddr=0x00020000
loadaddr_kernel=0x3080000
loadaddr_rtos=0x00001000
lock=10100000
loglevel=1
mac=90:0E:B3:22:2F:48
netmask=255.255.255.0
os_ident_addr=0x00500000
osd_reverse=0
otg_device=0
outputmode=1080p60hz
panel_type=lcd_1
partiton_mode=dynamic
preboot=run bcb_cmd; run upgrade_check;run init_display;run storeargs;run upgrade_key;run upgrade_adc_key;bcb uboot-command;run switch_bootmode;
reboot_mode=cold_boot
reboot_mode_android=normal
recovery_from_fat_dev=setenv loadaddr ${loadaddr_kernel};if fatload ${fatload_dev} 0 ${loadaddr} aml_autoscript; then autoscr ${loadaddr}; fi;if fatload ${fatload_dev} 0 ${loadaddr} recovery.img; then if fatload ${fatload_dev} 0 ${dtb_mem_addr} dtb.img; then echo ${fatload_dev} dtb.img loaded; fi;setenv bootargs ${bootargs} ${fs_type};bootm ${loadaddr};fi;
recovery_from_flash=get_valid_slot;echo active_slot: ${active_slot};setenv loadaddr ${loadaddr_kernel};if test ${active_slot} = normal; then setenv bootargs ${bootargs} ${fs_type} aml_dt=${aml_dt} recovery_part=${recovery_part} recovery_offset=${recovery_offset};if test ${upgrade_step} == 3; then if ext4load mmc 1:2 ${dtb_mem_addr} /recovery/dtb.img; then echo cache dtb.img loaded; fi;if ext4load mmc 1:2 ${loadaddr} /recovery/recovery.img; then echo cache recovery.img loaded; wipeisb; bootm ${loadaddr}; fi;else if imgread dtb recovery ${dtb_mem_addr}; then else echo restore dtb; run common_dtb_load;fi;fi;if imgread kernel ${recovery_part} ${loadaddr} ${recovery_offset}; then bootm ${loadaddr}; fi;else if fdt addr ${dtb_mem_addr}; then else echo retry common dtb; run common_dtb_load; fi;if test ${partiton_mode} = normal; then setenv bootargs ${bootargs} ${fs_type} aml_dt=${aml_dt} recovery_part=${boot_part} recovery_offset=${recovery_offset};if imgread kernel ${boot_part} ${loadaddr}; then bootm ${loadaddr}; fi;else if test ${vendor_boot_mode} = true; then setenv bootargs ${bootargs} ${fs_type} aml_dt=${aml_dt} recovery_part=${boot_part} recovery_offset=${recovery_offset} androidboot.slot_suffix=${active_slot};if imgread kernel ${boot_part} ${loadaddr}; then bootm ${loadaddr}; fi;else setenv bootargs ${bootargs} ${fs_type} aml_dt=${aml_dt} recovery_part=${recovery_part} recovery_offset=${recovery_offset} androidboot.slot_suffix=${active_slot};if imgread kernel ${recovery_part} ${loadaddr} ${recovery_offset}; then wipeisb; bootm ${loadaddr}; fi;fi;fi;fi;
recovery_from_sdcard=setenv fatload_dev mmc;if mmcinfo; then run recovery_from_fat_dev; fi;
recovery_from_udisk=setenv fatload_dev usb;if usb start 0; then run recovery_from_fat_dev; fi;
recovery_offset=0
recovery_part=recovery_a
region_code=US
sdcburncfg=aml_sdc_burn.ini
sddtb=if fatload mmc 0 ${dtb_mem_addr} dtb.img; then else store dtb read $dtb_mem_addr; fi
serial=1234567890
serial#=1234567890
serverip=10.18.9.113
silent=1
slot-suffixes=0
soc=sc2
stderr=serial@a000
stdin=serial@a000
stdout=serial@a000
storage_param=store param;setenv bootargs ${bootargs} ${mtdbootparts}; 
storeargs=setenv bootargs ${initargs} loglevel={loglevel} otg_device=${otg_device} logo=${display_layer},loaded,${fb_addr} vout=${outputmode},${vout_init} panel_type=${panel_type} hdmitx=${cecconfig},${colorattribute} hdmimode=${hdmimode} hdmichecksum=${hdmichecksum} dolby_vision_on=${dolby_vision_on} hdr_policy=${hdr_policy} hdr_priority=${hdr_priority} frac_rate_policy=${frac_rate_policy} hdmi_read_edid=${hdmi_read_edid} cvbsmode=${cvbsmode} osd_reverse=${osd_reverse} video_reverse=${video_reverse} irq_check_en=${Irq_check_en}  androidboot.selinux=${EnableSelinux} androidboot.firstboot=${firstboot} jtag=${jtag}; setenv bootargs ${bootargs} androidboot.hardware=amlogic;run cmdline_keys;
storeboot=run get_os_type;run storage_param;if test ${os_type} = rtos; then setenv loadaddr ${loadaddr_rtos};store read ${loadaddr} ${boot_part} 0 0x400000;bootm ${loadaddr};else if test ${os_type} = kernel; then get_system_as_root_mode;echo system_mode in storeboot: ${system_mode};get_valid_slot;get_avb_mode;echo active_slot in storeboot: ${active_slot};if test ${system_mode} = 1; then setenv bootargs ${bootargs} ro rootwait skip_initramfs;else setenv bootargs ${bootargs} androidboot.force_normal_boot=1;fi;if test ${active_slot} != normal; then setenv bootargs ${bootargs} androidboot.slot_suffix=${active_slot};fi;if fdt addr ${dtb_mem_addr}; then else echo retry common dtb; run common_dtb_load; fi;setenv loadaddr ${loadaddr_kernel};if imgread kernel ${boot_part} ${loadaddr}; then bootm ${loadaddr}; fi;else echo wrong OS format ${os_type}; fi;fi;echo try upgrade as booting failure; run update;
switch_bootmode=get_rebootmode;if test ${reboot_mode} = factory_reset; then run recovery_from_flash;else if test ${reboot_mode} = update; then run update;else if test ${reboot_mode} = quiescent; then setenv bootargs ${bootargs} androidboot.quiescent=1;else if test ${reboot_mode} = recovery_quiescent; then setenv bootargs ${bootargs} androidboot.quiescent=1;run recovery_from_flash;else if test ${reboot_mode} = cold_boot; then run usb_burning;else if test ${reboot_mode} = fastboot; then fastboot 0;fi;fi;fi;fi;fi;fi;
update=run usb_burning; run recovery_from_sdcard;run recovery_from_udisk;run recovery_from_flash;
upgrade_adc_key=saradc open 2; saradc getval; if saradc get_in_range 0 50; then echo update by adc-upgrade key...;run update;fi;
upgrade_check=echo recovery_status=${recovery_status};if itest.s "${recovery_status}" == "in_progress"; then run init_display;run storeargs; run recovery_from_flash;else fi;echo upgrade_step=${upgrade_step}; if itest ${upgrade_step} == 3; then run init_display;run storeargs; run update; fi;
upgrade_key=if gpio input GPIOD_3; then echo detect upgrade key; run update;fi;
upgrade_step=2
usb_burning=adnl 1200
usbdtb=if fatload usb 0 ${dtb_mem_addr} dtb.img; then else store dtb read $dtb_mem_addr; fi
usid=1234567890
vendor=amlogic
vendor_boot_mode=true
vendor_boot_part=vendor_boot
video_reverse=0
vout_init=disable -n bootcmd)
cmd_boot_sd=if mmcinfo; then setenv devtype mmc; setenv devnum 0; setenv distro_bootpart 1; run cmd_bootsript_load; fi
cmd_boot_usb=if usb storage; then ; else usb start; fi; setenv devtype usb; setenv distro_bootpart 1; for devnum in "0 1 2 3"; do setenv devnum ${devnum}; run cmd_bootsript_load; done
cmd_bootsript_load=for scrpt in "${boot_script_order}"; do fatload ${devtype} ${devnum}:${distro_bootpart} ${loadaddr} ${scrpt} && run cmd_make_compat && autoscr ${loadaddr}; done;
cmd_make_compat=cmd_read="fatload ${devtype} ${devnum}:${distro_bootpart}"; device=${devtype};devnr=${devnum};partnr=${distro_bootpart};partnum=${distro_bootpart};autoscript_source=${devtype}
cmdline_keys=setenv usid 1234567890; setenv region_code US;if keyman init 0x1234; then if keyman read usid ${loadaddr} str; then fi;if keyman read region_code ${loadaddr} str; then fi;if keyman read mac ${loadaddr} str; then setenv bootargs ${bootargs} mac=${mac} androidboot.mac=${mac};fi;if keyman read deviceid ${loadaddr} str; then setenv bootargs ${bootargs} androidboot.deviceid=${deviceid};fi;fi;setenv bootargs ${bootargs} androidboot.wificountrycode=${region_code};setenv bootargs ${bootargs} androidboot.serialno=${usid};setenv serial ${usid}; setenv serial# ${usid};factory_provision init;
colorattribute=444,8bit
common_dtb_load=imgread dtb boot_a 0x1000000
cpu=armv8
cvbs_drv=0
cvbsmode=576cvbs
default_boot_order=org usb sd emmc
display_bpp=16
display_color_bg=0
display_color_fg=0xffff
display_color_index=16
display_height=1080
display_layer=osd0
display_width=1920
dolby_status=0
dolby_vision_on=0
dtb_mem_addr=0x1000000
eth6addr=90:0e:b3:22:2f:48
fatload_dev=usb
fb_addr=0x00300000
fb_height=1080
fb_width=1920
fdt_high=0x20000000
fdtaddr=1000000
fdtcontroladdr=f077c700
firstboot=0
frac_rate_policy=1
fs_type=rootfstype=ramfs
gatewayip=10.18.9.1
get_os_type=if store read ${os_ident_addr} ${boot_part} 0 0x1000; then os_ident ${os_ident_addr}; fi
hdmi_colordepth=8
hdmi_colorspace=auto
hdmi_read_edid=1
hdmichecksum=0x3ff20000
hdmimode=1080p60hz
hdr_policy=1
hostname=arm_gxbb
init_display=get_rebootmode;echo reboot_mode:::: ${reboot_mode};if test ${reboot_mode} = quiescent; then setenv reboot_mode_android quiescent;setenv dolby_status 0;setenv dolby_vision_on 0;run storeargs;setenv bootargs ${bootargs} androidboot.quiescent=1;osd open;osd clear;else if test ${reboot_mode} = recovery_quiescent; then setenv reboot_mode_android quiescent;setenv dolby_status 0;setenv dolby_vision_on 0;run storeargs;setenv bootargs ${bootargs} androidboot.quiescent=1;osd open;osd clear;else setenv reboot_mode_android normal;run storeargs;hdmitx hpd;hdmitx get_preferred_mode;hdmitx get_parse_edid;setenv dolby_status 0;setenv dolby_vision_on 0;osd open;osd clear;imgread pic logo bootup $loadaddr;bmp display $bootup_offset;bmp scale;vout output ${outputmode};vpp hdrpkt;fi;fi;
initargs=init=/init console=ttyS0,921600 no_console_suspend earlycon=aml-uart,0xfe07a000 ramoops.pstore_en=1 ramoops.record_size=0x8000 ramoops.console_size=0x4000 loop.max_part=4 
initrd_high=df800000
ipaddr=10.18.9.97
is.bestmode=false
jtag=disable
loadaddr=0x00020000
loadaddr_kernel=0x3080000
loadaddr_rtos=0x00001000
lock=10100000
loglevel=1
mac=90:0E:B3:22:2F:48
netmask=255.255.255.0
os_ident_addr=0x00500000
osd_reverse=0
otg_device=0
outputmode=1080p60hz
panel_type=lcd_1
partiton_mode=dynamic
preboot=run bcb_cmd; run upgrade_check;run init_display;run storeargs;run upgrade_key;run upgrade_adc_key;bcb uboot-command;run switch_bootmode;
reboot_mode=cold_boot
reboot_mode_android=normal
recovery_from_fat_dev=setenv loadaddr ${loadaddr_kernel};if fatload ${fatload_dev} 0 ${loadaddr} aml_autoscript; then autoscr ${loadaddr}; fi;if fatload ${fatload_dev} 0 ${loadaddr} recovery.img; then if fatload ${fatload_dev} 0 ${dtb_mem_addr} dtb.img; then echo ${fatload_dev} dtb.img loaded; fi;setenv bootargs ${bootargs} ${fs_type};bootm ${loadaddr};fi;
recovery_from_flash=get_valid_slot;echo active_slot: ${active_slot};setenv loadaddr ${loadaddr_kernel};if test ${active_slot} = normal; then setenv bootargs ${bootargs} ${fs_type} aml_dt=${aml_dt} recovery_part=${recovery_part} recovery_offset=${recovery_offset};if test ${upgrade_step} == 3; then if ext4load mmc 1:2 ${dtb_mem_addr} /recovery/dtb.img; then echo cache dtb.img loaded; fi;if ext4load mmc 1:2 ${loadaddr} /recovery/recovery.img; then echo cache recovery.img loaded; wipeisb; bootm ${loadaddr}; fi;else if imgread dtb recovery ${dtb_mem_addr}; then else echo restore dtb; run common_dtb_load;fi;fi;if imgread kernel ${recovery_part} ${loadaddr} ${recovery_offset}; then bootm ${loadaddr}; fi;else if fdt addr ${dtb_mem_addr}; then else echo retry common dtb; run common_dtb_load; fi;if test ${partiton_mode} = normal; then setenv bootargs ${bootargs} ${fs_type} aml_dt=${aml_dt} recovery_part=${boot_part} recovery_offset=${recovery_offset};if imgread kernel ${boot_part} ${loadaddr}; then bootm ${loadaddr}; fi;else if test ${vendor_boot_mode} = true; then setenv bootargs ${bootargs} ${fs_type} aml_dt=${aml_dt} recovery_part=${boot_part} recovery_offset=${recovery_offset} androidboot.slot_suffix=${active_slot};if imgread kernel ${boot_part} ${loadaddr}; then bootm ${loadaddr}; fi;else setenv bootargs ${bootargs} ${fs_type} aml_dt=${aml_dt} recovery_part=${recovery_part} recovery_offset=${recovery_offset} androidboot.slot_suffix=${active_slot};if imgread kernel ${recovery_part} ${loadaddr} ${recovery_offset}; then wipeisb; bootm ${loadaddr}; fi;fi;fi;fi;
recovery_from_sdcard=setenv fatload_dev mmc;if mmcinfo; then run recovery_from_fat_dev; fi;
recovery_from_udisk=setenv fatload_dev usb;if usb start 0; then run recovery_from_fat_dev; fi;
recovery_offset=0
recovery_part=recovery_a
region_code=US
sdcburncfg=aml_sdc_burn.ini
sddtb=if fatload mmc 0 ${dtb_mem_addr} dtb.img; then else store dtb read $dtb_mem_addr; fi
serial=1234567890
serial#=1234567890
serverip=10.18.9.113
silent=1
slot-suffixes=0
soc=sc2
stderr=serial@a000
stdin=serial@a000
stdout=serial@a000
storage_param=store param;setenv bootargs ${bootargs} ${mtdbootparts}; 
storeargs=setenv bootargs ${initargs} loglevel={loglevel} otg_device=${otg_device} logo=${display_layer},loaded,${fb_addr} vout=${outputmode},${vout_init} panel_type=${panel_type} hdmitx=${cecconfig},${colorattribute} hdmimode=${hdmimode} hdmichecksum=${hdmichecksum} dolby_vision_on=${dolby_vision_on} hdr_policy=${hdr_policy} hdr_priority=${hdr_priority} frac_rate_policy=${frac_rate_policy} hdmi_read_edid=${hdmi_read_edid} cvbsmode=${cvbsmode} osd_reverse=${osd_reverse} video_reverse=${video_reverse} irq_check_en=${Irq_check_en}  androidboot.selinux=${EnableSelinux} androidboot.firstboot=${firstboot} jtag=${jtag}; setenv bootargs ${bootargs} androidboot.hardware=amlogic;run cmdline_keys;
storeboot=run get_os_type;run storage_param;if test ${os_type} = rtos; then setenv loadaddr ${loadaddr_rtos};store read ${loadaddr} ${boot_part} 0 0x400000;bootm ${loadaddr};else if test ${os_type} = kernel; then get_system_as_root_mode;echo system_mode in storeboot: ${system_mode};get_valid_slot;get_avb_mode;echo active_slot in storeboot: ${active_slot};if test ${system_mode} = 1; then setenv bootargs ${bootargs} ro rootwait skip_initramfs;else setenv bootargs ${bootargs} androidboot.force_normal_boot=1;fi;if test ${active_slot} != normal; then setenv bootargs ${bootargs} androidboot.slot_suffix=${active_slot};fi;if fdt addr ${dtb_mem_addr}; then else echo retry common dtb; run common_dtb_load; fi;setenv loadaddr ${loadaddr_kernel};if imgread kernel ${boot_part} ${loadaddr}; then bootm ${loadaddr}; fi;else echo wrong OS format ${os_type}; fi;fi;echo try upgrade as booting failure; run update;
switch_bootmode=get_rebootmode;if test ${reboot_mode} = factory_reset; then run recovery_from_flash;else if test ${reboot_mode} = update; then run update;else if test ${reboot_mode} = quiescent; then setenv bootargs ${bootargs} androidboot.quiescent=1;else if test ${reboot_mode} = recovery_quiescent; then setenv bootargs ${bootargs} androidboot.quiescent=1;run recovery_from_flash;else if test ${reboot_mode} = cold_boot; then run usb_burning;else if test ${reboot_mode} = fastboot; then fastboot 0;fi;fi;fi;fi;fi;fi;
update=run usb_burning; run recovery_from_sdcard;run recovery_from_udisk;run recovery_from_flash;
upgrade_adc_key=saradc open 2; saradc getval; if saradc get_in_range 0 50; then echo update by adc-upgrade key...;run update;fi;
upgrade_check=echo recovery_status=${recovery_status};if itest.s "${recovery_status}" == "in_progress"; then run init_display;run storeargs; run recovery_from_flash;else fi;echo upgrade_step=${upgrade_step}; if itest ${upgrade_step} == 3; then run init_display;run storeargs; run update; fi;
upgrade_key=if gpio input GPIOD_3; then echo detect upgrade key; run update;fi;
upgrade_step=2
usb_burning=adnl 1200
usbdtb=if fatload usb 0 ${dtb_mem_addr} dtb.img; then else store dtb read $dtb_mem_addr; fi
usid=1234567890
vendor=amlogic
vendor_boot_mode=true
vendor_boot_part=vendor_boot
video_reverse=0
vout_init=disable

**Reply by devmfc** [reply]
I don't see really disturbing things that could block USB Burn Tool.  It should just work. I can't reproduce it on my X96 X4 box either.

But maybe you can try to boot to  Debian and execute the following two commands: 

`fw_setenv cmd_boot_org 'run storeboot'`
`fw_setenv default_boot_order 'sd usb emmc org'`

But I don't expect that it will work. But no harm in trying...

**Reply by Kullsputra** [reply]
Hi,
Idk why the reset button wasn't forcing the box into "flashing mode" to use with AML Burn Tool via USB in the PC.
The emmc short point I was using before was actually working as the reset as well, I tried other points in the PCB to short the emmc and it worked, the box was detected in the PC (via USB 3.0 port) and I was able to flash Android back into it.
I haven't tried the Debian commands you wrote about, sorry.

Thanks for all the help!

**Comment by Kullsputra** [comment]
For future reference to other people, I used this image as reference to find the emmc short points in the X96_X4:
You have to connect the green points (one by one) to ground (USB port shield, for ex) and test until it works.
![24129634](https://github.com/user-attachments/assets/ca51f8eb-489d-4fde-b87f-60e513e5d9d1)

In my case I tried this point in the first try and it worked right away:
![241296d34](https://github.com/user-attachments/assets/a9704e6d-0d19-49b6-927e-f4260738e17f)

I used AML USB Burn Tool V3.2.8.
The procedure was: I shorted the point in image to ground with a wire and held it by hand, then connected the box to the PC via the USB 3.0 port.
Nothing else was connected.

**Reply by Kullsputra** [reply]
Update: The reset A/V button is now working again to put the box in USB flash mode.

**Reply by devmfc** [reply]
Good to hear, thanx for the images!

**Reply by Kullsputra** [reply]
No problem, glad to help where I can!
